spring-cloud-aws-autoconfigure/src/main/java/org/springframework/cloud/aws/autoconfigure/context/ContextCredentialsAutoConfiguration.java

```diff
@@ -85,6 +94,27 @@ public AWSCredentialsProvider awsCredentialsProvider(
 					: new ProfileCredentialsProvider(properties.getProfileName()));
 		}
 
+		AwsCredentialsProperties.StsCredentialsProperties stsProperties = properties
+				.getSts();
+		if (StringUtils.hasText(stsProperties.getRoleArn())
+				&& StringUtils.hasText(stsProperties.getRoleSessionName())) {
+
+			AWSSecurityTokenServiceClientBuilder stsClientBuilder = AWSSecurityTokenServiceClientBuilder
```

Comment:
Since `sts` is an optional dependency, if user provides `roleArn` and `roleSessionName` but does not add dependency to `sts` code will fail with `NoClassDefFoundException`. 

We have to change it in a way that there is an attempt to use STS only if `sts` related classes are on the classpath.

Reply:
logic for `sts` should be in another class with the proper `Conditional*` annotations and import it in this auto-configuration class just using the`AWSCredentialProvider`. In order to test it we can use `ClasspathExclusions` from `spring-boot` like [this example](https://github.com/spring-projects/spring-boot/blob/8e695b0cc5e834a01e9f12fa35af3b7fddddf58a/spring-boot-project/spring-boot/src/test/java/org/springframework/boot/web/servlet/NoSpringWebFilterRegistrationBeanTests.java)